src/visitable.cpp

```diff
@@ -829,6 +829,21 @@ long visitable<Character>::charges_of( const std::string &what, long limit ) con
         }
     }
 
+    if( what == "breath" ) {
+        long qty = 0;
+        if( p && p->has_item_with_flag( "AIR_SUPPLY" ) ) {
+            for( auto &i : p->all_items_with_flag( "AIR_SUPPLY" ) ) {
+                qty = sum_no_wrap( qty, charges_of( i->typeId() ) );
```

Comment:
This may count items several times: imagine a player with two SCUBA items (of same type) in their inventory. The loop will go through both of them (`all_items_with_flag` returns two entries), so this line will be executed twice. `i->typeId()` will be the same each time, so you are adding "all charges of items of type 'X'" twice.

Why are you not using the charges of the items you iterate over directly?

Also: why are you checking the existence of items with that flag before iteration of them? What do you expect to happen upon calling `all_items_with_flag` without any suitable items in the inventory? It will just return an empty set and the loop body will not be run at all. The `has_item_with_flag` is completely redundant.



Same problems in the code below.